tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/require-expect"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester(),
+    returnAndIndent = "\n        ",
+    ALWAYS_ERROR_MESSAGE = "Should use `assert.expect` when using assertions outside of the top-level test callback",
+    EXCEPT_SIMPLE_ERROR_MESSAGE = "Test is missing `assert.expect() call`";
+
+ruleTester.run("require-expect", rule, {
+
+    valid: [
+        // default, calling expect is valid
+        {
+            code: "test('name', function(assert) { assert.expect(0) });",
+            options: []
+        },
```

Comment:
I mentioned in the docs that it would be good to have a few examples of global `expect`. Unfortunately, in the unit tests, I have to ask for equal coverage of global and non-global `expect`. Sorry!